### DISCUSSION TITLE
Add Timestamp Query Parameter to List Branches for Head Commit

### DISCUSSION THREAD

**Discussion by nkreiger** [community/community#63839] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

For this REST API Endpoint: https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-branches-for-head-commit

It would be really nice to allow to search at a point in time to be able to reproduce results at a later point, and also, because its the only quick way to associate a commit to a branch without retrieving a list, and performing pagination, as far as I know. `?timestamp=`

**Comment by Mronyx** [comment]
To add a timestamp query parameter to list branches for the head commit in a programming language, you would typically need to make an HTTP request to the API endpoint that provides the list of branches. Here's an example in Python using the requests library:
import requests
import datetime

url = 'https://api.github.com/repos/{owner}/{repo}/branches'
timestamp = datetime.datetime.now().timestamp() # Get the current timestamp

# Add the timestamp as a query parameter
params = {
    'head': 'master',
    'timestamp': timestamp
}

response = requests.get(url, params=params)

# Check if the request was successful
if response.status_code == 200:
    branches = response.json()  # Retrieve the branches as JSON
    # Process the branches as needed
    print(branches)
else:
    print('Request failed with status code:', response.status_code)
Make sure to replace {owner} and {repo} with the respective values for your repository. Additionally, you may need to include any required authentication headers depending on the API's authentication requirements.

This code snippet sends a GET request to the GitHub API's endpoint for listing branches, with the head parameter set to "master" (or any other branch name you specify) and the timestamp parameter set to the current timestamp. The response is then processed accordingly, based on the response status code.
